Geometry_Engine/Modify/Offset.cs

```diff
@@ -0,0 +1,868 @@
+
+                            List<Point> curveIntersecions = pCurve1.ICurveIntersections(pCurve2, tolerance);
+                            if (curveIntersecions.Count > 0)
+                                intersection = curveIntersecions.ClosestPoint(curve1.IStartPoint());
+                            else
+                            {
+                                Reflection.Compute.RecordWarning("Couldn't create fillet");
+                                return null;
+                            }
+                            PolyCurve subResult1 = new PolyCurve { Curves = curve1.ExtendToPoint(intersection, curve1.IEndPoint(), tangentExtensions, tolerance).ToList() };
+                            PolyCurve subResult2 = new PolyCurve { Curves = curve2.ExtendToPoint(curve2.IStartPoint(), intersection, tangentExtensions, tolerance).ToList() };
+                            subResult1 = subResult1.Flip();
+                            subResult2 = subResult2.Flip();
+                            resultCurves.AddRange(subResult1.Curves);
+                            resultCurves.AddRange(subResult2.Curves);
+                        }
+                        else
+                        {
+                            Line tanLine1 = Create.Line(curve1.IStartPoint(), curve1.IStartDir().Reverse() / tolerance);
+                            tanLine1.Infinite = false;
+                            PolyCurve pCurve1 = new PolyCurve { Curves = { tanLine1, curve1 } };
+
+                            Line tanLine2 = Create.Line(curve2.IStartPoint(), curve2.IStartDir().Reverse() / tolerance);
+                            tanLine2.Infinite = false;
+                            PolyCurve pCurve2 = new PolyCurve { Curves = { tanLine2, curve2 } };
+
+                            List<Point> curveIntersecions = pCurve1.ICurveIntersections(pCurve2, tolerance);
+                            if (curveIntersecions.Count > 0)
+                                intersection = curveIntersecions.ClosestPoint(curve1.IStartPoint());
+                            else
+                            {
+                                Reflection.Compute.RecordWarning("Couldn't create fillet");
+                                return null;
+                            }
+                            PolyCurve subResult1 = new PolyCurve { Curves = curve1.ExtendToPoint(intersection, curve1.IEndPoint(), tangentExtensions, tolerance).ToList() };
+                            PolyCurve subResult2 = new PolyCurve { Curves = curve2.ExtendToPoint(intersection, curve2.IEndPoint(), tangentExtensions, tolerance).ToList() };
+                            subResult1 = subResult1.Flip();
+                            resultCurves.AddRange(subResult1.Curves);
+                            resultCurves.AddRange(subResult2.Curves);
+                        }
+                    }
+                }
+            }
+            for (int i = 0; i < resultCurves.Count; i++)
```

Comment:
Shouldn't this loop iterate backwards?

Plus a blank line before the `for` loop please.